core/src/main/resources/error/error-classes.json

```diff
@@ -11,6 +11,25 @@
     "message" : [ "Found duplicate keys '%s'" ],
     "sqlState" : "23000"
   },
+  "GROUPING_COLUMN_MISMATCH" : {
+    "message" : [ "Column of grouping (%s) can't be found in grouping columns %s" ],
+    "sqlState" : "22023"
```

Comment:
sorry @karenfeng if I missed but where do we document these SQL states?

Reply:
The error class README contains a section discussing SQLSTATE: https://github.com/apache/spark/blob/master/core/src/main/resources/error/README.md, including this specific rule:
```
Spark does not define its own classes or subclasses.
```
The SQLSTATEs are determined by consulting the SQL standard.

Reply:
In this case, all of the new error classes fall under SQLSTATE 22023, which represents an invalid parameter value. Another common SQLSTATE is 42000, which represents all syntax and semantic errors.

Reply:
Actually, for these grouping PRs, it will be easier for developers to leave them out for now; we can always add them down the line.